clyent/colors/color_formatter.py

```diff
@@ -34,39 +34,102 @@ def get_field(self, field_name, args, kwargs):
         else:
             return string.Formatter.get_field(self, field_name, args, kwargs)
 
-
-    def _vformat(self, format_string, args, kwargs, used_args, recursion_depth):
-        if recursion_depth < 0:
-            raise ValueError('Max string recursion exceeded')
-        result = []
-
-        for literal_text, field_name, format_spec, conversion in \
-                self.parse(format_string):
-
-            # output the literal text
-            self.stream.write(literal_text)
-
-            # if there's a field, output it
-            if field_name is not None:
-                # this is some markup, find the object and do
-                #  the formatting
-
-                # given the field_name, find the object it references
-                #  and the argument it came from
-                obj, arg_used = self.get_field(field_name, args, kwargs)
-                used_args.add(arg_used)
-
-                # do any conversion on the resulting object
-                obj = self.convert_field(obj, conversion)
-
-                # expand the format spec, if needed
-                format_spec = string.Formatter()._vformat(format_spec, args, kwargs,
-                                                          used_args, recursion_depth - 1)
-
-                # format the object and append to the result
-                self.format_field(obj, format_spec)
-
-        return ''.join(result)
+    # The signature of _vformat changed in Python 3.4. It added a new argument
+    # `auto_arg_index`.
+    # https://github.com/python/cpython/blob/3.4/Lib/string.py#L192
+    # https://github.com/python/cpython/blob/2.7/Lib/string.py#L567
+    if sys.version_info[:2] > (3, 3):
+        def _vformat(self, format_string, args, kwargs, used_args, recursion_depth,
+                     auto_arg_index=0):
```

Comment:
Just noticing this, would it be possible to have some helper function that holds the content of this function and then is called by one of these two signatures? Mainly concern about divergence of the two branches and more maintenance work.